In [2]:
import pandas as pd

In [3]:
data = pd.read_csv("hazard_reformatted_2_19.csv")

/var/folders/p6/tqnlcwnd075_z06_lxt45pv40000gn/T/ipykernel_22895/255799653.py:1: DtypeWarning: Columns (6,11) have mixed types. Specify dtype option on import or set low_memory=False.
  data = pd.read_csv("hazard_reformatted_2_19.csv")


In [4]:
data.head()

,Unnamed: 0.1,Product Id,Company,Brand,Product Name,Variant,Product Discontinued Date,Product Submitted Date,Ingredient Name,Function,...,Hepatotoxicity And Digestive System Toxicity,Immunotoxicity,Musculoskeletal Toxicity,Nephrotoxicity And Other Toxicity To The Urinary System,Neurodevelopmental Toxicity,Neurotoxicity,Ocular Toxicity,Other Toxicological Hazard Traits,Reproductive Toxicity,Respiratory Toxicity
0,0,18358,Nail Alliance - Entity,Entity Nudite,Cool Pink Nail Sculpting Powder,NaN,NaN,6/24/2019,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,...,False,False,False,False,False,False,False,False,False,False
1,1,38661,"Xtreme Color, Inc.",Hard Candy,Fierce Effects- Shadow Duo,Black-Solid,NaN,9/21/2015,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,...,False,False,False,False,False,False,False,False,False,False
2,2,38662,"Xtreme Color, Inc.",Hard Candy,Fierce Effects- Shadow Duo,Black-Sprinkle,NaN,9/21/2015,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,...,False,False,False,False,False,False,False,False,False,False
3,3,38667,"Xtreme Color, Inc.",Hard Candy,Fierce Effects- Shadow Duo,Brown-Solid,NaN,9/21/2015,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,...,False,False,False,False,False,False,False,False,False,False
4,4,38665,"Xtreme Color, Inc.",Hard Candy,Fierce Effects- Shadow Duo,Brown-Solid,NaN,9/21/2015,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,...,False,False,False,False,False,False,False,False,False,False


In [21]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 534514 entries, 0 to 534513
Data columns (total 46 columns):
 #   Column                                                   Non-Null Count   Dtype  
---  ------                                                   --------------   -----  
 0   Unnamed: 0.1                                             534514 non-null  int64  
 1   Product Id                                               534514 non-null  int64  
 2   Company                                                  534514 non-null  object 
 3   Brand                                                    534314 non-null  object 
 4   Product Name                                             534514 non-null  object 
 5   Variant                                                  187814 non-null  object 
 6   Product Discontinued Date                                39104 non-null   object 
 7   Product Submitted Date                                   534514 non-null  object 
 8   Ingredient Nam

unique ingredients, multiple companies

In [22]:
data["Product Submitted Year"] = pd.to_datetime(data["Product Submitted Date"]).dt.year
data["Product Discontinued Year"] = pd.to_datetime(data["Product Discontinued Date"]).dt.year
data.head()

,Unnamed: 0.1,Product Id,Company,Brand,Product Name,Variant,Product Discontinued Date,Product Submitted Date,Ingredient Name,Function,...,Musculoskeletal Toxicity,Nephrotoxicity And Other Toxicity To The Urinary System,Neurodevelopmental Toxicity,Neurotoxicity,Ocular Toxicity,Other Toxicological Hazard Traits,Reproductive Toxicity,Respiratory Toxicity,Product Submitted Year,Product Discontinued Year
0,0,18358,Nail Alliance - Entity,Entity Nudite,Cool Pink Nail Sculpting Powder,NaN,NaN,6/24/2019,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,...,False,False,False,False,False,False,False,False,2019,NaN
1,1,38661,"Xtreme Color, Inc.",Hard Candy,Fierce Effects- Shadow Duo,Black-Solid,NaN,9/21/2015,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,...,False,False,False,False,False,False,False,False,2015,NaN
2,2,38662,"Xtreme Color, Inc.",Hard Candy,Fierce Effects- Shadow Duo,Black-Sprinkle,NaN,9/21/2015,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,...,False,False,False,False,False,False,False,False,2015,NaN
3,3,38667,"Xtreme Color, Inc.",Hard Candy,Fierce Effects- Shadow Duo,Brown-Solid,NaN,9/21/2015,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,...,False,False,False,False,False,False,False,False,2015,NaN
4,4,38665,"Xtreme Color, Inc.",Hard Candy,Fierce Effects- Shadow Duo,Brown-Solid,NaN,9/21/2015,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,...,False,False,False,False,False,False,False,False,2015,NaN


In [15]:
data["Company"].nunique()

1396

1,396 companies have reported

In [ ]:
unique_companies = data.groupby("Company")["Product Id"].nunique().rename("count").reset_index()
unique_companies.sort_values("count", ascending = False)
#this is companies by the number of products they've submitted

,Company,count
693,L'Oreal USA,9120
121,Bath & Body Works,7007
1224,The Procter & Gamble Company (Cincinnati),4947
277,Coty,4466
1051,Revlon Consumer Product Corporation,3722
...,...,...
1228,The Tend Skin Company,1
797,"Mad Rabbit Tattoo, Inc.",1
439,Foamology,1
167,Bonjout Beauty,1


In [33]:
submission_year = data.groupby("Product Submitted Year")["Product Id"].nunique().rename("count").reset_index()
submission_year

,Product Submitted Year,count
0,2009,22794
1,2010,11303
2,2011,3411
3,2012,2911
4,2013,4847
5,2014,6906
6,2015,5675
7,2016,3529
8,2017,4847
9,2018,6928


In [31]:
import altair as alt

In [32]:
alt.Chart(submission_year).mark_bar().encode(
    x = "Product Submitted Year",
    y = "count"
)

alt.Chart(...)

In [14]:
data["Brand"].nunique()

4448

4,448 brands have reported

In [5]:
len(data)

534514

NEED TO MERGE PROD NAME AND VARIANT LIKELY TO ANALYZE PROPERLY

In [6]:
data["Brand"].nunique()

4448

4,448 brands have reported their products to CDPH since X YEAR

In [13]:
data["Product Id"].nunique()

165503

165,503 products have been reported to CDPH since X YEAR

In [ ]:
ingreds = data.groupby("Ingredient Name").size().rename("count").reset_index()
ingreds.sort_values("count", ascending = False)

,Ingredient Name,count
451,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,91510
311,Limonene (1-methyl-4-prop-1-en-2-yl-cyclohexen...,42034
10,"1,6-Octadien-3-ol, 3,7-dimethyl- (LINALOOL) 78...",29596
125,Benzyl alcohol 100-51-6,17246
128,Benzyl salicylate 118-58-1,15145
...,...,...
313,Magnesium silicate talc,1
114,"Benzenesulfonic acid, 5-amino-2,4-dimethyl-, d...",1
117,Benzo[a]anthracene (Benz[a]anthracene) 56-55-3,1
118,Benzo[a]pyrene (BaP) (Benzo[def]chrysene; Benz...,1


In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 534514 entries, 0 to 534513
Data columns (total 46 columns):
 #   Column                                                   Non-Null Count   Dtype  
---  ------                                                   --------------   -----  
 0   Unnamed: 0.1                                             534514 non-null  int64  
 1   Product Id                                               534514 non-null  int64  
 2   Company                                                  534514 non-null  object 
 3   Brand                                                    534314 non-null  object 
 4   Product Name                                             534514 non-null  object 
 5   Variant                                                  187814 non-null  object 
 6   Product Discontinued Date                                39104 non-null   object 
 7   Product Submitted Date                                   534514 non-null  object 
 8   Ingredient Nam

In [ ]:
data["Product Discontinued Date"] = pd.to_datetime(data['Product Discontinued Date'])
data['Product Discontinued Year'] = data['Product Discontinued Date'].dt.year
data["Product Submitted Date"] = pd.to_datetime(data['Product Submitted Date'])
data['Product Submitted Year'] = data['Product Submitted Date'].dt.year
data["Ingredient Submitted Date"] = pd.to_datetime(data['Ingredient Submitted Date'])
data['Ingredient Submitted Year'] = data['Ingredient Submitted Date'].dt.year
data["Ingredient Removed Date"] = pd.to_datetime(data['Ingredient Removed Date'])
data['Ingredient Removed Year'] = data['Ingredient Removed Date'].dt.year

In [ ]:
data.head()

,Unnamed: 0.1,Product Id,Company,Brand,Product Name,Variant,Product Discontinued Date,Product Submitted Date,Ingredient Name,Function,Unit of Measure,Concentration,Ingredient Submitted Date,Ingredient Removed Date,cas_number,clean_name,api_rn,api_name,Hazard Traits,Chemical Name,Synonyms,Authoritative List,"When to Report (Any, FF, FA)",Unnamed: 0,Bioaccumulation,Carcinogenicity,Cardiovascular Toxicity,Dermatotoxicity,Developmental Toxicity,Endocrine Toxicity,Environmental Persistence,Environmental Tox,Fragrance Allergen,Genotoxicity,Hazard Trait Undefined,Hematotoxicity,Hepatotoxicity And Digestive System Toxicity,Immunotoxicity,Musculoskeletal Toxicity,Nephrotoxicity And Other Toxicity To The Urinary System,Neurodevelopmental Toxicity,Neurotoxicity,Ocular Toxicity,Other Toxicological Hazard Traits,Reproductive Toxicity,Respiratory Toxicity,Product Discontinued Year,Product Submitted Year,Ingredient Submitted Year,Ingredient Removed Year
0,0,18358,Nail Alliance - Entity,Entity Nudite,Cool Pink Nail Sculpting Powder,NaN,NaT,2019-06-24,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,mg/g,5.000000,2019-06-24,NaT,13463-67-7,Titanium dioxide (CI 77891),13463-67-7,Titania,Carcinogenicity,"Titanium dioxide (airborne, unbound particles ...",CI 77891,IARC Carcinogens - 2B; Prop 65,Any,NaN,False,True,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,NaN,2019,2019.0,NaN
1,1,38661,"Xtreme Color, Inc.",Hard Candy,Fierce Effects- Shadow Duo,Black-Solid,NaT,2015-09-21,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,mg/g,40.000000,2015-09-21,NaT,13463-67-7,Titanium dioxide (CI 77891),13463-67-7,Titania,Carcinogenicity,"Titanium dioxide (airborne, unbound particles ...",CI 77891,IARC Carcinogens - 2B; Prop 65,Any,NaN,False,True,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,NaN,2015,2015.0,NaN
2,2,38662,"Xtreme Color, Inc.",Hard Candy,Fierce Effects- Shadow Duo,Black-Sprinkle,NaT,2015-09-21,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,mg/g,31.000000,2015-09-21,NaT,13463-67-7,Titanium dioxide (CI 77891),13463-67-7,Titania,Carcinogenicity,"Titanium dioxide (airborne, unbound particles ...",CI 77891,IARC Carcinogens - 2B; Prop 65,Any,NaN,False,True,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,NaN,2015,2015.0,NaN
3,3,38667,"Xtreme Color, Inc.",Hard Candy,Fierce Effects- Shadow Duo,Brown-Solid,NaT,2015-09-21,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,mg/g,12.000000,2015-09-21,NaT,13463-67-7,Titanium dioxide (CI 77891),13463-67-7,Titania,Carcinogenicity,"Titanium dioxide (airborne, unbound particles ...",CI 77891,IARC Carcinogens - 2B; Prop 65,Any,NaN,False,True,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,NaN,2015,2015.0,NaN
4,4,38665,"Xtreme Color, Inc.",Hard Candy,Fierce Effects- Shadow Duo,Brown-Solid,NaT,2015-09-21,Titanium dioxide (CI 77891) 13463-67-7 / 1317-...,NaN,mg/g,21.500000,2015-09-21,NaT,13463-67-7,Titanium dioxide (CI 77891),13463-67-7,Titania,Carcinogenicity,"Titanium dioxide (airborne, unbound particles ...",CI 77891,IARC Carcinogens - 2B; Prop 65,Any,NaN,False,True,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,NaN,2015,2015.0,NaN
